index.bs

```diff
@@ -124,22 +131,78 @@ A conversion report is a [=struct=] with the following items:
 
 # Algorithms # {#algorithms}
 
-<h3 algorithm id="creating-conversion">Creating a conversion</h3>
-TODO
+<h3 algorithm id="parsing-conversion-destination">Parsing a conversion destination</h3>
 
-<h3 algorithm id="creating-impression">Creating an impression</h3>
-TODO
+To <dfn>parse a conversion destination</dfn> from an <{a}> tag |anchor|,
+1. Let |url| be the result of running the [=URL parser=] on the value of
+    the |anchor|'s <{a/conversiondestination}>.
+1. Return the result of [=obtain a site|obtaining a site=] from |url|'s
+    [=url/origin=].
+
+<h3 algorithm id="creating-impression">Activating an impression</h3>
+
+To <dfn>activate an impression</dfn> from an <{a}> tag |anchor|,
+1. Let |currentTime| be the current time.
+1. Let |impression| be a new [=impression=] struct whose items are:
+
+: [=impression/impression source=]
+:: |anchor|'s [=relevant settings object=]'s [=environment/top-level origin=].
+: [=impression/impression data=]
+:: The result of applying [=parsing conversion data=] to |anchor|'s
+    <{a/impressiondata}> attribute.
+: [=impression/conversion destination=]
+:: The result of running [=parse a conversion destination=] on |anchor|.
+: [=impression/reporting endpoint=]
+:: The [=url/origin=] of the result of running the [=URL parser=] on the value
+    of |anchor|'s <{a/reportingorigin}> attribute.
+: [=impression/expiry=]
+:: |currentTime| +  <{a/impressionexpiry}> milliseconds.
+: [=impression/impression time=]
+:: |currentTime|.
```

Comment:
I think you can indent all of this by 4 spaces in order to make it part of the list item. That'll become important when you add subsequent list items to store the impression.

Reply:
Done

Reply:
Done